core/services/eth/log_broadcaster_test.go

```diff
@@ -318,3 +318,61 @@ func TestDecodingLogListener(t *testing.T) {
 	listener.HandleLog(nil, expectedErr)
 	require.Equal(t, err, expectedErr)
 }
+
+func TestLogBroadcaster_ReceivesAllLogsWhenResubscribing(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	const blockHeight uint64 = 0
+
+	ethClient := new(mocks.Client)
+	sub := new(mocks.Subscription)
+
+	chBlock := make(chan struct{})
+	chchRawLogs := make(chan chan<- eth.Log, 1)
+	var unblock struct{}
+
+	ethClient.On("SubscribeToLogs", mock.Anything, mock.Anything, mock.Anything).
+		Run(func(args mock.Arguments) {
+			chRawLogs := args.Get(1).(chan<- eth.Log)
+			chchRawLogs <- chRawLogs
+			<-chBlock
+			return
+		}).
+		Return(sub, nil).
+		Twice()
+
+	ethClient.On("GetBlockHeight").Return(blockHeight, nil)
+	sub.On("Err").Return(nil)
+	sub.On("Unsubscribe").Return()
+
+	lb := ethsvc.NewLogBroadcaster(ethClient, store.ORM)
+	lb.Start()
+
+	logCount := 0
+	logListener := funcLogListener{
+		fn: func(log interface{}, err error) { logCount++ },
+	}
+	logListener2 := funcLogListener{
+		fn: func(log interface{}, err error) {},
```

Comment:
I think we might want to test the number of logs received by this listener as well, just to be on the safe side.

Reply:
It feels kind of pointless to me, because the second logListener is listening for contract events at a different address (it has to be different or else we don't actually run the resubscribe loop) but the events all belong to `Address{0}`. So the count for the second log listener should always ever be 0, and if that breaks somehow it should absolutely be breaking other tests. So I'd prefer to just keep this simpler.